content/blog/return0e/2017-08-28_[gsoc_2017]_porting_swift_to_haiku_-_final_report.md

```diff
@@ -0,0 +1,45 @@
++++
+type = "blog"
+title = "[GSoC 2017] Porting Swift to Haiku - Final Report"
+author = "return0e"
+date = "2017-08-28 07:15:32+01:00"
+tags = ["haiku", "swift", "gsoc17","software"]
++++
+
+Hello everyone!
+
+This blog-post marks the final report on bringing Swift to Haiku in the Google Summer of Code period. My introductory post on this project can be found [here](https://www.haiku-os.org/blog/return0e/2017-05-10_gsoc_2017_porting_the_swift_programming_language_to_haiku) for a brief overview of the project.
+
+## Summary
+In the last 3 coding periods, my contributions to Haiku's LLVM and Clang ports plus reporting some bugs with the Haiku developers have made it possible for the Swift toolchain to be built on Haiku. With this, it opens the possibility to use cross-platform Swift libraries used on other platforms and also allows to directly use the libc/glibc libraries via the GlibC module. I have already done an initial port of Foundation and libdispatch on Haiku as specified in the previous blog-post, but they still need to be polished for general use. As for upstreaming my patches for Haiku support, I've sent my patches to apple/swift and they are currently under [review](https://github.com/apple/swift/pull/11583).
+
+Here's a short review below on what has been completed and still outstanding on this project and the current [documentation](https://gist.github.com/return/c9e28df2ea0009310835e51996a379c9)  to get Swift building on Haiku.
+
+## What has been completed:
+* Building swiftc and its standard library.
+* Execute simple swift programs (compiled or interpreted).
+* Support for integrated REPL (non-LLDB version) works
+* Initial port of corelibs-foundation and libdispatch.
+* Run tests against the newly built standard library and swiftc.
+* Swift 3.1 recipe available on HaikuPorts.
```

Comment:
A list URLs pointing to these completed items would be nice, ie https://github.com/return/swift-corelibs-libdispatch/commits/swift-3.1-haiku

Reply:
@korli For some completed tasks, could a link to the blog post that mentions this task e.g [Integrated REPL support](https://www.haiku-os.org/blog/return0e/2017-07-28_gsoc_2017_porting_swift_to_haiku_-_week_8/) be acceptable?

Reply:
For me who knows a bit the project it's already difficult to follow and find the parts you worked on. Blog posts are interesting but don't help to summarize where patches you produced reside.
Keep in mind that's also a deliverable for GSoC (probably in a zip file though).

Reply:
@korli I'm very sorry about the lack of information on my progress. I've recently documented my progress in a [github gist](https://gist.github.com/return/c9e28df2ea0009310835e51996a379c9) if that helps to clear up the confusion.

As for the deliverables, most of my work is in the upstream [PR](https://github.com/apple/swift/pull/11583),  So I think that creating a zip that has a script that re-creates the recent build environment with the documentation should acceptable for work submission.

Reply:
maybe just add the content of your github gist in the swift haikuports PR.